pypeit/wavecalib.py

```diff
@@ -413,15 +597,20 @@ def update_wvmask(self):
 
         """
         # Update mask based on wv_calib
-        for key in self.wv_calib.keys():
-            if key in ['steps', 'par', 'fit2d', 'bpm']:
-                continue
-            if (self.wv_calib[key] is None) or (len(self.wv_calib[key]) == 0):
-                try:
-                    idx = self.slits.spatid_to_zero(int(key))
-                except:
-                    embed(header='428 of wavecalib')
-                self.wvc_bpm[idx] = True
+        #for key in self.wv_calib.keys():
+        #    if key in ['steps', 'par', 'fit2d', 'bpm']:
+        #        continue
+        #    if (self.wv_calib[key] is None) or (len(self.wv_calib[key]) == 0):
+        #        try:
+        #            idx = self.slits.spatid_to_zero(int(key))
+        #        except:
+        #            embed(header='428 of wavecalib')
+        #        self.wvc_bpm[idx] = True
+        #
+        for kk, fit in enumerate(self.wv_calib.wv_fits):
+            if fit is None:
+                self.wvc_bpm[kk] = True
```

Comment:
If you don't need the old bpm:
```
self.wvc_bpm = np.array([fit is None for fit in self.wv_calib.wv_fits])
```

If you do:
```
self.wvc_bpm |= np.array([fit is None for fit in self.wv_calib.wv_fits])
```

Reply:
am too scared :)